sunpy/net/dataretriever/sources/noaa.py

```diff
@@ -200,6 +158,21 @@ def _get_url_for_timerange(self, timerange, **kwargs):
             result.append(url)
         return result
 
+    def search(self, *args, **kwargs):
+        extractor1 = '{}/warehouse/{:4d}/SRS/{year:4d}{month:2d}{day:2d}SRS.txt'
+        extractor2 = '{}/warehouse/{year:4d}/{}'
+        matchdict = self._get_match_dict(*args, **kwargs)
+        timerange = matchdict['Time']
+        metalist = []
+        for url in self._get_url_for_timerange(timerange):
+            exdict1 = parse(extractor1, url)
+            exdict2 = parse(extractor2, url)
+            exdict = (exdict2 if exdict1 is None else exdict1).named
+            map_ = super().post_search_hook(exdict, matchdict)
```

Comment:
why `super()` we haven't overriden it right?

Reply:
Actually the `SRSClient` is weird. It changes its pattern on the basis of year. So I first know the correct pattern and then use the `super().search` cause it saves me from adding scraper code.

Reply:
My point  was `self.post_search_hook` should be fine.

Reply:
Oh ok, corrected.